worker/request_handlers.go

```diff
@@ -70,24 +75,49 @@ func HandleSelfStreamRecordEnd(ctx *StreamContext) {
 	notifySilenceResults(sd.Silences, ctx.streamId)
 }
 
-//HandleSelfStreamEnd stops the ffmpeg instance by sending a SIGINT to it and prevents the loop to restart it by marking the stream context as stopped.
-func HandleSelfStreamEnd(ctx *StreamContext) {
+func HandleStreamEndRequest(request *pb.EndStreamRequest) {
+	log.Println("Attempting to end stream: ", request.StreamID)
+	streams := lectureHallStreams[request.StreamID]
+	for streamIndex := range streams {
+		streamCtx := streams[streamIndex]
+		if streamCtx.isSelfStream {
+			log.Error("Unexpected self stream end request.")
+			continue
+		}
+		HandleStreamEnd(streamCtx)
+	}
+}
+
+//HandleStreamEnd stops the ffmpeg instance by sending a SIGINT to it and prevents the loop to restart it by marking the stream context as stopped.
+func HandleStreamEnd(ctx *StreamContext) {
 	ctx.stopped = true
 	if ctx.streamCmd != nil && ctx.streamCmd.Process != nil {
 		err := ctx.streamCmd.Process.Kill()
 		if err != nil {
-			log.WithError(err).Warn("can't kill self-stream ffmpeg")
+			log.WithError(err).Warn("can't kill ffmpeg process")
 		}
 	} else {
-		log.Warn("self-stream context has no command on stream end")
+		log.Warn("context has no command or process to end")
 	}
 	S.endStream(ctx)
 	notifyStreamDone(ctx.streamId)
+	lectureHallStreams[ctx.streamId] = remove(lectureHallStreams[ctx.streamId], ctx)
+}
+
+func remove(contexts []*StreamContext, context *StreamContext) []*StreamContext {
+	var newContexts []*StreamContext
+
+	for _, i := range contexts {
+		if i.sourceUrl != context.sourceUrl {
+			newContexts = append(newContexts, i)
+		}
+	}
+	log.Println("Removed stream with source: ", context.sourceUrl)
```

Comment:
Can you use `Info` or `Trace` instead of `Println`?

Reply:
I actually used that just for debugging, but now that I think about it, this should be fairly useful.